File: clients/typescript/src/exceptions/ApiError.ts
import type { Problem } from "../models/Problem.js";

/** Base class for all errors thrown by the API client. */
export class ApiError extends Error {
  constructor(
    public readonly status: number,
    public readonly problem: Problem,
  ) {
    super(problem.detail || problem.type);
    this.name = "ApiError";
  }
}

File: clients/swift/src/Exceptions/ApiError.swift
import Foundation

/// Base class for all HTTP-status errors returned by the API.
open class ApiError: Error, @unchecked Sendable {
    public let status: Int
    public let problem: Problem

    public init(status: Int, problem: Problem) {
        self.status = status
        self.problem = problem
    }
}
